src/python.rs

```diff
@@ -164,9 +186,176 @@ fn forward(
     }
 }
 
-// TODO: Mock
+/// mocks the prover
+#[pyfunction(signature = (
+    data,
+    model,
+    py_run_args = None
+))]
+fn mock(
+    data: String,
+    model: String,
+    py_run_args: Option<PyRunArgs>
+) -> Result<bool, PyErr> {
+    let run_args: RunArgs = py_run_args.unwrap_or_else(PyRunArgs::new).into();
+    let logrows = run_args.logrows;
+    let data = prepare_data(data);
+    let visibility = run_args.to_var_visibility();
+
+    match data {
+        Ok(d) => {
+            let procmodel = Model::<Fr>::new(model, run_args, Mode::Mock, visibility);
+
+            match procmodel {
```

Comment:
Could we clean this up and replace each match with something like 
```rust 
 let procmodel = Model::<Fr>::new(model, run_args, Mode::Prove, visibility);
 // replace with 
  let procmodel = Model::<Fr>::new(model, run_args, Mode::Prove, visibility).map_err(|_| PyIOError::new_err("Failed to process model"))?; 
```

Reply:
Let me try this, will get back soon!